src/profile/ResetPassword.js

```diff
@@ -1,148 +1,251 @@
   };
 
   const renderRequestCodeForm = () => (
-    <form onSubmit={handleSendCodeClick}>
-      <FormGroup bsSize="large" controlId="email">
-        <ControlLabel>{I18n.get("email")}</ControlLabel>
-        <FormControl
-          autoFocus
-          type="email"
-          value={state.email}
-          onChange={handleChange}
-        />
-      </FormGroup>
-      <LoaderButton
-        block
-        type="submit"
-        size="large"
-        loadingText={I18n.get("sending")}
-        text={I18n.get("sendConfirmation")}
-        isLoading={state.isSendingCode}
-        disabled={!validateEmailForm()}
-      />
-    </form>
+    <div className="Form">
+      <form
+        className={classes.root}
+        onSubmit={handleSubmit(handleSendCodeClick)}
+      >
+        <div>
+          <TextField
+            id="email"
+            variant="filled"
+            size="medium"
+            autoFocus
+            label={I18n.get("email")}
+            {...register("email", {
+              required: "email is required",
+              pattern: {
+                value: /^[^\s@]+@([^\s@.,]+\.)+[^\s@.,]{2,}$/,
+                message: "invalid email address",
+              },
+            })}
+          />
+        </div>
+        <span className="error">{errors.email && errors.email.message}</span>
+        <div>
+          <LoaderButton
+            loadingText={I18n.get("sending")} //
+            text={I18n.get("sendConfirmation")} //
+            isLoading={isSendingCode} //
+            disabled={disabled} //
+            type="submit"
+            color="primary"
+            variant="contained"
+          />
+        </div>
+      </form>
+    </div>
   );
 
   const renderConfirmationForm = () => (
-    <form onSubmit={handleConfirmClick}>
-      <FormGroup bsSize="large" controlId="code">
-        <ControlLabel>{I18n.get("confirmationCode")}</ControlLabel>
-        <FormControl
-          autoFocus
-          type="tel"
-          value={state.code}
-          onChange={handleChange}
-        />
-        <HelpBlock>{I18n.get("checkEmail")}</HelpBlock>
-      </FormGroup>
-      <hr />
-      <FormGroup bsSize="large" controlId="password">
-        <ControlLabel>{I18n.get("newPassword")}</ControlLabel>
-        <FormControl
-          type="password"
-          value={state.password}
-          onChange={handleChange}
-        />
-      </FormGroup>
-      <FormGroup bsSize="large" controlId="confirmPassword">
-        <ControlLabel>{I18n.get("confirm")}</ControlLabel>
-        <FormControl
-          type="password"
-          onChange={handleChange}
-          value={state.confirmPassword}
-        />
-      </FormGroup>
-      <LoaderButton
-        block
-        type="submit"
-        size="large"
-        text={I18n.get("confirm")}
-        loadingText={I18n.get("confirming")}
-        isLoading={state.isConfirming}
-        disabled={!validateResetForm()}
-      />
-    </form>
+    <div className="Form">
+      <Container>
```

Comment:
Can you remove this unnecessary Container element? Or explain why it's needed?

Reply:
Hi again, so when changing between the email entry page and the code confirmation page, when I take the container tag out, the input from the email entry gets carried over to the code confirmation TextField when the code confirmation form is rendered. Not sure why this is, but adding the container tag seems to fix this. If you can suggest another way to fix this issue that would be faster I can do that.

Reply:
I replaced this with a div instead which worked just as well, so the container element doesn't need to be imported